Make more use of UniqueRef in WebCore::Page

youennf opened the pull request:
#### 969004bb4fc5be727a58339a770c363c2173d57c
<pre>
Make more use of UniqueRef in WebCore::Page
<a href="https://bugs.webkit.org/show_bug.cgi?id=254513">https://bugs.webkit.org/show_bug.cgi?id=254513</a>
rdar://problem/107265298

Reviewed by Alex Christensen.

Replace some const unique_ptr by UniqueRef in Page class, to ease reading of getters.

* Source/WebCore/page/Chrome.h:
(WebCore::Chrome::client const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::remoteInspectorInformationDidChange):
(WebCore::Page::remoteInspectorInformationDidChange const): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Page::chrome):
(WebCore::Page::chrome const):
(WebCore::Page::dragCaretController):
(WebCore::Page::dragCaretController const):
(WebCore::Page::dragController):
(WebCore::Page::dragController const):
(WebCore::Page::contextMenuController):
(WebCore::Page::contextMenuController const):
(WebCore::Page::userInputBridge):
(WebCore::Page::inspectorController):
(WebCore::Page::pointerCaptureController):
(WebCore::Page::pointerLockController):
(WebCore::Page::progress):
(WebCore::Page::progress const):
(WebCore::Page::backForward):
(WebCore::Page::console):
(WebCore::Page::console const):
(WebCore::Page::inspectorDebuggable):
(WebCore::Page::inspectorDebuggable const):
(WebCore::Page::userInputBridge const): Deleted.
(WebCore::Page::inspectorController const): Deleted.
(WebCore::Page::pointerCaptureController const): Deleted.
(WebCore::Page::pointerLockController const): Deleted.
(WebCore::Page::backForward const): Deleted.
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::sanitizeLookalikeCharactersIfNeeded):

Canonical link: <a href="https://commits.webkit.org/262262@main">https://commits.webkit.org/262262@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f969cd191da346e83f7c80d34e5ac4c598e7dadc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/898 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/922 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/957 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/1274 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/794 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/886 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/972 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1007 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/998 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/906 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/846 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/841 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1216 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/899 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/836 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/839 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/813 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/862 "8 failures") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/1868 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/854 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/807 "4 failures") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/801 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/847 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/256 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/868 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/page/Page.cpp:
```diff
@@ -278,24 +278,24 @@ static Ref<Frame> createMainFrame(Page& page, std::variant<UniqueRef<FrameLoader
+    , m_dragController(makeUniqueRef<DragController>(*this, WTFMove(pageConfiguration.dragClient)))
 #endif
-    , m_focusController(makeUnique<FocusController>(*this, pageInitialActivityState()))
+    , m_focusController(std::make_unique<FocusController>(*this, pageInitialActivityState()))
```
This should be makeUnique to check that it is fast malloc'ed.

youennf replied:
done

achristensen07 commented on Source/WebCore/page/Page.h:
```diff
@@ -1079,28 +1086,28 @@ class Page : public Supplementable<Page>, public CanMakeWeakPtr<Page> {
 
-    const std::unique_ptr<Chrome> m_chrome;
-    const std::unique_ptr<DragCaretController> m_dragCaretController;
+    UniqueRef<Chrome> m_chrome;
```
I'm a little sad that these can't stay const, indicating that the object can't be changed, while keeping the const correct model of UniqueRef, which only allows you to get a non-const object reference from it if it is not const.  I think this is still an improvement, though.

Nit: T* ptr() const in UniqueRef.h should be const T* ptr() const

youennf replied:
right, let's do that in a follow up

youennf committed:
Make more use of UniqueRef in WebCore::Page
https://bugs.webkit.org/show_bug.cgi?id=254513
rdar://problem/107265298

Reviewed by Alex Christensen.

Replace some const unique_ptr by UniqueRef in Page class, to ease reading of getters.

* Source/WebCore/page/Chrome.h:
(WebCore::Chrome::client const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::Page):
(WebCore::Page::remoteInspectorInformationDidChange):
(WebCore::Page::remoteInspectorInformationDidChange const): Deleted.
* Source/WebCore/page/Page.h:
(WebCore::Page::chrome):
(WebCore::Page::chrome const):
(WebCore::Page::dragCaretController):
(WebCore::Page::dragCaretController const):
(WebCore::Page::dragController):
(WebCore::Page::dragController const):
(WebCore::Page::contextMenuController):
(WebCore::Page::contextMenuController const):
(WebCore::Page::userInputBridge):
(WebCore::Page::inspectorController):
(WebCore::Page::pointerCaptureController):
(WebCore::Page::pointerLockController):
(WebCore::Page::progress):
(WebCore::Page::progress const):
(WebCore::Page::backForward):
(WebCore::Page::console):
(WebCore::Page::console const):
(WebCore::Page::inspectorDebuggable):
(WebCore::Page::inspectorDebuggable const):
(WebCore::Page::userInputBridge const): Deleted.
(WebCore::Page::inspectorController const): Deleted.
(WebCore::Page::pointerCaptureController const): Deleted.
(WebCore::Page::pointerLockController const): Deleted.
(WebCore::Page::backForward const): Deleted.
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::sanitizeLookalikeCharactersIfNeeded):

Canonical link: https://commits.webkit.org/262262@main
